### PR TITLE
Fix `BlobGasUsed` and `ExcessBlobGas` in Cancun genesis

### PR DESCRIPTION
**Pull request by marcindsobczak**
## Changes

- post cancun, default `BlobGasUsed` and `ExcessBlobGas` to 0 if not specified in genesis
- it is fixing many failing hive consensus tests

#### What types of changes does your code introduce?

- [x] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

## Testing

- [x] Yes
- [ ] No

#### If yes, did you write tests?

- [ ] Yes
- [x] No

#### Notes on testing

- tested using hive tests

#### Requires documentation update

- [ ] Yes
- [x] No

#### Requires explanation in Release Notes

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/Nethermind/Nethermind.Specs/ChainSpecStyle/ChainSpecLoader.cs [hunk 1]
```diff
@@ -423,8 +423,8 @@ private static void LoadGenesis(ChainSpecJson chainSpecJson, ChainSpec chainSpec
         bool isEip4844Enabled = chainSpecJson.Params.Eip4844TransitionTimestamp != null && genesisHeader.Timestamp >= chainSpecJson.Params.Eip4844TransitionTimestamp;
         if (isEip4844Enabled)
         {
-            genesisHeader.BlobGasUsed = chainSpecJson.Genesis.BlobGasUsed;
-            genesisHeader.ExcessBlobGas = chainSpecJson.Genesis.ExcessBlobGas;
+            genesisHeader.BlobGasUsed = chainSpecJson.Genesis.BlobGasUsed ?? 0;
+            genesisHeader.ExcessBlobGas = chainSpecJson.Genesis.ExcessBlobGas ?? 0;
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Specs/ChainSpecStyle/ChainSpecLoader.cs:427]: simpler would be to change ChainSpecGenesisJson and make those properties non-nullable - they would default to 0

Comment (marcindsobczak) [reply src/Nethermind/Nethermind.Specs/ChainSpecStyle/ChainSpecLoader.cs:427]: From what I understand, pre-cancun we expect it to be null, so we need it nullable. In this case genesis is with cancun EIPs already enabled so we need it as 0 when calculating hash

Comment (marcindsobczak) [reply src/Nethermind/Nethermind.Specs/ChainSpecStyle/ChainSpecLoader.cs:427]: so if eip4844 is disabled when loading genesis, it is null
https://github.com/NethermindEth/nethermind/blob/master/src/Nethermind/Nethermind.Specs/ChainSpecStyle/ChainSpecLoader.cs#L423-L428

Comment (marcindsobczak) [reply src/Nethermind/Nethermind.Specs/ChainSpecStyle/ChainSpecLoader.cs:427]: and we just need it to be 0 if genesis is post-cancun and value is not specified in genesis

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Specs/ChainSpecStyle/ChainSpecLoader.cs:427]: make it non-nullable in `chainSpecJson.Genesis.BlobGasUsed` so in `ChainSpecGenesisJson`

Comment (marcindsobczak) [reply src/Nethermind/Nethermind.Specs/ChainSpecStyle/ChainSpecLoader.cs:427]: good point. Changing

**Review by LukaszRozmej — APPROVED** [review]

**Review by MarekM25 — APPROVED** [review]

**Review by MarekM25 — APPROVED** [review]

**Pull request merged**